openbsd/OpenBSDProcessList.c

```diff
@@ -190,22 +192,35 @@ static char* OpenBSDProcessList_readProcessName(kvm_t* kd, const struct kinfo_pr
    /* don't use xMalloc here - we want to handle huge argv's gracefully */
    char* s;
    if ((s = malloc(len)) == NULL) {
-      *basenameEnd = strlen(kproc->p_comm);
-      return xStrdup(kproc->p_comm);
+      Process_updateCmdline(proc, kproc->p_comm, 0, strlen(kproc->p_comm));
+      return;
    }
 
    *s = '\0';
 
+   int start = 0;
+   int end = 0;
    for (int i = 0; arg[i] != NULL; i++) {
       size_t n = strlcat(s, arg[i], len);
       if (i == 0) {
-         *basenameEnd = MINIMUM(n, len - 1);
+         end = MINIMUM(n, len - 1);
+         for (int j = end; j > 0; j--) {
+            if (arg[0][j] == ' ' && arg[0][j-1] != '\\') {
+               end = (arg[0][j-1] == ':') ? (j-1) : j;
+            }
+         }
+         for (int j = end - 2; j > 0; j--) {
+            if (arg[0][j] == '/' && arg[0][j-1] != '\\') {
+               start = j+1;
+               break;
+            }
+         }
```

Comment:
Some descriptive comments for this would be good.

Reply:
Added a comment for the first loop, the second one is now obsolete due to the above changes